### PR TITLE
Updated search endpoint to return retrieve URLs

### PR DESCRIPTION
**Pull request by emmberk**
- Injected the retrieve endpoint into the ingest service
- Updated the /search endpoint to return a list of retrieve URLs instead of the id and cst

### CODE REVIEW

**Comment by emmberk** [comment]
rebased

File: ingest/src/main/java/com/connexta/ingest/service/impl/IngestServiceImpl.java [hunk 1]
```diff
@@ -29,16 +30,20 @@
...
   @NotNull private final TransformClient transformClient;
-  private final String callbackEndpoint;
+  @NotEmpty private final String callbackEndpoint;
+  @NotEmpty private final String retrieveEndpoint;
 
   public IngestServiceImpl(
       @NotNull final S3StorageAdaptor s3Adaptor,
       @NotNull final TransformClient transformClient,
-      @Value("${endpointUrl.ingest.callback}") String callbackEndpoint) {
+      @NotEmpty @Value("${endpointUrl.ingest.callback}") final String callbackEndpoint,
```
Comment (jhunzik) [inline ingest/src/main/java/com/connexta/ingest/service/impl/IngestServiceImpl.java]: It doesn't have to be in this PR, but we should move the property injection out to the config file. Similar to how we do it in the S3Adaptor. This keeps our Spring stuff separated from our business logic.

File: multi-int-store/src/main/java/com/connexta/multiintstore/services/impl/SearchServiceImpl.java [hunk 2]
```diff
@@ -9,23 +9,42 @@
...
 public class SearchServiceImpl implements SearchService {
 
-  private final IndexedMetadataRepository indexedMetadataRepository;
+  @NotNull private final IndexedMetadataRepository indexedMetadataRepository;
+
+  @NotEmpty private final String retrieveEndpoint;
+
+  public SearchServiceImpl(
+      @NotNull final IndexedMetadataRepository indexedMetadataRepository,
+      @NotEmpty @Value("${endpointUrl.ingest.retrieve}") final String retrieveEndpoint) {
```
Comment (jhunzik) [inline multi-int-store/src/main/java/com/connexta/multiintstore/services/impl/SearchServiceImpl.java]: Same as previous comment

File: ingest/src/main/java/com/connexta/ingest/service/impl/IngestServiceImpl.java (hunk 1 above)
Comment (emmberk) [reply ingest/src/main/java/com/connexta/ingest/service/impl/IngestServiceImpl.java]: :+1: Let's address that in a different PR.

**Comment by emmberk** [comment]
rebased

**Comment by emmberk** [comment]
`./gradlew clean build` and `./gradlew dependencyCheckAnalyze --info` succeeded locally for me

**Comment by emmberk** [comment]
I manually heroed this

**Comment by emmberk** [comment]
build now

**Comment by emmberk** [comment]
build now

**Comment by emmberk** [comment]
The CI build failed because of a known issue downloading CVEs. I'm overriding that build check because OWASP succeeded locally.

**Pull request merged**